cmd/open.go

```diff
@@ -15,17 +25,155 @@ var openCmd = &cobra.Command{
 
 Pass the path to the directory that contains the solution you want to see on the website.
 	`,
-	Args: cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg := config.NewConfig()
+
+		v := viper.New()
+		v.AddConfigPath(cfg.Dir)
+		v.SetConfigName("user")
+		v.SetConfigType("json")
+		// Ignore error. If the file doesn't exist, that is fine.
+		_ = v.ReadInConfig()
+		cfg.UserViperConfig = v
+
+		return runOpen(cfg, cmd.Flags(), args)
+	},
+}
+
+func runOpen(cfg config.Config, flags *pflag.FlagSet, args []string) error {
+	var url string
+	usrCfg := cfg.UserViperConfig
+	trackID, _ := flags.GetString("track")
+	exerciseID, _ := flags.GetString("exercise")
+	teamID, _ := flags.GetString("team")
+
+	if exerciseID == "" {
+		// if no --exercise is given, use original functionality
 		metadata, err := workspace.NewExerciseMetadata(args[0])
 		if err != nil {
 			return err
 		}
 		browser.Open(metadata.URL)
 		return nil
-	},
+		//return fmt.Errorf("Must provide an `--exercise`")
+	}
+
+	if remote, _ := flags.GetBool("remote"); remote {
+		apiUrl := fmt.Sprintf("%s/solutions/latest", usrCfg.GetString("apibaseurl"))
+
+		client, err := api.NewClient(usrCfg.GetString("token"), usrCfg.GetString("apibaseurl"))
+		if err != nil {
+			return err
+		}
+
+		req, err := client.NewRequest("GET", apiUrl, nil)
+		if err != nil {
+			return err
+		}
+
+		q := req.URL.Query()
+		q.Add("exercise_id", exerciseID)
+		if trackID != "" {
+			q.Add("track_id", trackID)
+		}
+		if teamID != "" {
+			q.Add("team_id", teamID)
+		}
+		req.URL.RawQuery = q.Encode()
+
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+
+		defer res.Body.Close()
+		var payload openPayload
+		if err := json.NewDecoder(res.Body).Decode(&payload); err != nil {
+			return fmt.Errorf("unable to parse API response - %s", err)
+		}
+
+		if res.StatusCode != http.StatusOK {
+			switch payload.Error.Type {
+			case "track_ambiguous":
```

Comment:
I wonder if we can make the error decoder helper smarter and add this logic into it.
https://github.com/exercism/cli/blob/c3525cf31135f24a84f8127f553e90b539b0298d/cmd/cmd.go#L78

(If so, let's do that in a separate PR and get that into master quickly so we can use it elsewhere).